### PR TITLE
initial data script

### PR DESCRIPTION
**Pull request by shounakb1**
I have made the data creation script, this is a dirty implementation though. The way I make it work is by running:

First update the test.yaml with the train video and corresponding audio. Then run:

python -m scripts.data --inference_config configs/inference/test.yaml --folder_name train

Again update the test.yaml with test video and audio and again running: 

python -m scripts.data --inference_config configs/inference/test.yaml --folder_name test

This creates folders which contain the image frames and npy files.

I also need to modify the Dataloader.py in train_codes to make the training hapen.
I can make the necessary changes before you merge @czk32611 @itechmusic .

https://github.com/TMElyralab/MuseTalk/assets/32771603/e3bba83f-93c3-4908-9213-9da596c39c49

Also attaching a sample to show the trained results, I cant upload more than 10MB so its small.

Just guide me on what changes to make and I'll do it. Thanks for this guys.

### CODE REVIEW

**Comment by czk32611** [comment]
Can you also show how should we modify `Dataloader.py`?

**Comment by shounakb1** [comment]
Hi @czk32611 , @itechmusic , I have commited the changes needed to make training and inference(with finetuned model) work.
Please check the README.md in train_codes.
Let me know anything which needs to be changed.

**Comment by shounakb1** [comment]
@czk32611  , @itechmusic There is no way to compare validation loss with the training loss right now in the train.py script. Do you think we should add that? I can add it if needed.

**Comment by paulovasconcellos-hotmart** [comment]
Can I ask what data you trained your model on?

**Comment by shounakb1** [comment]
> Can I ask what data you trained your model on?

The training data was youtube videos

**Comment by paulovasconcellos-hotmart** [comment]
Thanks for the quick reply! How many hours did your dataset have?

**Comment by shounakb1** [comment]
Just 15 mins fo this one, I trined on 1 hr data for better results

**Comment by paulovasconcellos-hotmart** [comment]
@shounakb1 I'm trying to use your code to train on multiple videos. When you have multiple files, the code won't work. Especially in [this part](https://github.com/TMElyralab/MuseTalk/blob/d74c4c098b3af834d795e0c8494214f92df5363a/train_codes/DataLoader.py#L96), where the code will get a reference image. If you have more videos in the folder, It might get a reference frame from another video.

**Comment by czk32611** [comment]
> @shounakb1 I'm trying to use your code to train on multiple videos. When you have multiple files, the code won't work. Especially in [this part](https://github.com/TMElyralab/MuseTalk/blob/d74c4c098b3af834d795e0c8494214f92df5363a/train_codes/DataLoader.py#L96), where the code will get a reference image. If you have more videos in the folder, It might get a reference frame from another video.

@shounakb1 paulovasconcellos-hotmart is right, data from each video should be saved to individual folder. Could you modify this part?

**Comment by shounakb1** [comment]
> @shounakb1 I'm trying to use your code to train on multiple videos. When you have multiple files, the code won't work. Especially in [this part](https://github.com/TMElyralab/MuseTalk/blob/d74c4c098b3af834d795e0c8494214f92df5363a/train_codes/DataLoader.py#L96), where the code will get a reference image. If you have more videos in the folder, It might get a reference frame from another video.



> > @shounakb1 I'm trying to use your code to train on multiple videos. When you have multiple files, the code won't work. Especially in [this part](https://github.com/TMElyralab/MuseTalk/blob/d74c4c098b3af834d795e0c8494214f92df5363a/train_codes/DataLoader.py#L96), where the code will get a reference image. If you have more videos in the folder, It might get a reference frame from another video.
> 
> @shounakb1 paulovasconcellos-hotmart is right, data from each video should be saved to individual folder. Could you modify this part?

Yes thanks for pointing it out guys, I'll fix it, sorry for the delay, work schedule is a bit hectic right now.

**Comment by shounakb1** [comment]
@czk32611 I've fixed the multiple video data preperation probem. Please let me know if anything else needs to be changed.
